python/lsst/analysis/drp/scatterPlot.py

```diff
@@ -0,0 +1,532 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from scipy.stats import median_absolute_deviation as sigmaMad
+from matplotlib import gridspec
+from matplotlib.patches import Rectangle
+from matplotlib.path import Path
+from matplotlib.collections import PatchCollection
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+from lsst.pipe.tasks.configurableActions import ConfigurableActionStructField
+from lsst.pipe.tasks.dataFrameActions import MagColumnNanoJansky, SingleColumnAction
+from lsst.skymap import BaseSkyMap
+
+import lsst.pipe.base as pipeBase
+import lsst.pex.config as pexConfig
+
+from . import dataSelectors as dataSelectors
+from .plotUtils import generateSummaryStats, parsePlotInfo, addPlotInfo
+
+
+class ScatterPlotWithTwoHistsTaskConnections(pipeBase.PipelineTaskConnections,
+                                             dimensions=("tract", "skymap"),
+                                             defaultTemplates={"inputCoaddName": "deep",
+                                                               "plotName": "deltaCoords"}):
+
+    catPlot = pipeBase.connectionTypes.Input(doc="The tract wide catalog to make plots from.",
+                                             storageClass="DataFrame",
+                                             name="objectTable_tract",
+                                             dimensions=("tract", "skymap"),
+                                             deferLoad=True)
+
+    skymap = pipeBase.connectionTypes.Input(doc="The skymap for the tract",
+                                            storageClass="SkyMap",
+                                            name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+                                            dimensions=("skymap",))
+
+    scatterPlot = pipeBase.connectionTypes.Output(doc="A scatter plot with histograms for both axes.",
+                                                  storageClass="Plot",
+                                                  name="scatterTwoHistPlot_{plotName}",
+                                                  dimensions=("tract", "skymap"))
+
+
+class ScatterPlotWithTwoHistsTaskConfig(pipeBase.PipelineTaskConfig,
+                                        pipelineConnections=ScatterPlotWithTwoHistsTaskConnections):
+
+    axisActions = ConfigurableActionStructField(
+        doc="The actions to use to calculate the values used on each axis.",
```

Comment:
Maybe explain here what the "mag" axis is?  `xAction` and `yAction` seem self-explanatory, but I don't understand off the bat what `magAction` is.

Reply:
Also seems like the fact that there is actually a default xAction should be mentioned in the doc here, since right now it's just hidden in the `setDefaults` method.

Reply:
The mag axis is used to decide on the points which go into the statistics.